bot/exts/halloween/candy_collection.py

```diff
@@ -77,33 +77,27 @@ async def on_reaction_add(self, reaction: discord.Reaction, user: discord.Member
                 await self.reacted_msg_chance(message)
             return
 
-        for react in self.msg_reacted:
-            # check to see if the message id of a message we added a
-            # reaction to is in json file, and if nobody has won/claimed it yet
-            if react['msg_id'] == message.id and react['won'] is False:
-                react['user_reacted'] = user.id
-                react['won'] = True
-                try:
-                    # if they have record/candies in json already it will do this
-                    user_records = self.get_candyinfo[user.id]
-                    if str(reaction.emoji) == '\N{CANDY}':
-                        user_records['record'] += 1
-                    if str(reaction.emoji) == '\N{SKULL}':
-                        if user_records['record'] <= 3:
-                            user_records['record'] = 0
-                            lost = 'all of your'
-                        else:
-                            lost = random.randint(1, 3)
-                            user_records['record'] -= lost
-                        await self.send_spook_msg(message.author, message.channel, lost)
-
-                except KeyError:
-                    # otherwise it will raise KeyError so we need to add them to file
-                    if str(reaction.emoji) == '\N{CANDY}':
-                        print('ok')
-                        d = {"userid": user.id, "record": 1}
-                        self.candy_json['records'].append(d)
-                await self.remove_reactions(reaction)
+        if message.id in self.candy_messages and str(reaction.emoji) == '\N{CANDY}':
+            self.candy_messages.remove(message.id)
+            prev_record = self.candy_records.get(str(message.author.id)) or 0
```

Comment:
Again here, use the default value argument of get() method
```suggestion
            prev_record = self.candy_records.get(str(message.author.id), 0)
```
Do this in all places.